packages/next/build/analysis/get-page-static-info.ts

```diff
@@ -313,17 +313,13 @@ export async function getPageStaticInfo(params: {
       !isEdgeRuntime(resolvedRuntime)
     ) {
       const options = Object.values(SERVER_RUNTIME).join(', ')
-      if (typeof resolvedRuntime !== 'string') {
-        Log.error(
-          `The \`runtime\` config must be a string. Please leave it empty or choose one of: ${options}`
-        )
-      } else {
-        Log.error(
-          `Provided runtime "${resolvedRuntime}" is not supported. Please leave it empty or choose one of: ${options}`
-        )
-      }
+      const message =
+        typeof resolvedRuntime !== 'string'
+          ? `The \`runtime\` config must be a string. Please leave it empty or choose one of: ${options}`
+          : `Provided runtime "${resolvedRuntime}" is not supported. Please leave it empty or choose one of: ${options}`
+      Log.error(message)
       if (!isDev) {
-        process.exit(1)
+        throw new Error(message)
```

Comment:
Why do we want to throw here instead of `process.exit(1)` I don't think showing the error stack here will be helpful for the user since it's internal modules.

Reply:
In general to have the process only exit in one place (the top level handler).

Reply:
Hmm we probably need to update the flushing for traces then possibly by wrapping `process.exit()` as we are already using the `process.exit(1)` pattern throughout to prevent the un-necessary stack traces showing which ultimately adds confusion instead of clarity. 

We can look at this more in a follow-up.